client/src/components/menubar/index.tsx

```diff
@@ -84,9 +90,53 @@ interface State {
     clipPercentileMax: number | undefined;
   };
 }
-class MenuBar extends React.PureComponent<MenuBarProps, State> {
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-  static isValidDigitKeyEvent(e: any) {
+
+const DIFFEXP_WIDTH = 150;
+const BUTTON_WIDTH = 38;
+
+function MenuBar(props: MenuBarProps) {
+  const {
+    dispatch,
+    disableDiffexp,
+    clipPercentileMin,
+    clipPercentileMax,
+    graphInteractionMode,
+    showCentroidLabels,
+    categoricalSelection,
+    colorAccessor,
+    subsetPossible,
+    subsetResetPossible,
+    screenCap,
+    imageUnderlay,
+  } = props;
+
+  const [pendingClipPercentiles, setPendingClipPercentiles] = useState<
+    State["pendingClipPercentiles"]
+  >({
+    clipPercentileMin: undefined,
+    clipPercentileMax: undefined,
+  });
+
+  const [widthUnits, setWidthUnits] = useState(8);
```

Comment:
by widthUnits do you mean how many buttons can fit in a row? If so, maybe `availableButtonSlots` conveys the idea 👌 ?

Reply:
I used a more general widthUnit, because some buttons span 2 units. I can switch if that makes more sense to you though!

Reply:
Ah good point that some buttons span two units 💡 ChatGPT has a few options:

availableButtonSlots
buttonSlotCount
buttonFitCount
buttonsPerRow
maxButtonCapacity
buttonSpaceCount
buttonSlotsAvailable
buttonWidthUnits

Maybe the key is to have `button` in the state name? Maybe you can pick one from above?